src/praisonai/praisonai/bots/telegram.py

```
@@ -534,6 +538,17 @@ async def _tg_unreact(emoji, **kw):
                                         except Exception as e:
                                             logger.error(f"Failed to send media: {e}")
                             
+                            # Render any agent-attached portable presentation
+                            # (buttons/menus) natively after the streamed text.
+                            try:
+                                presentation = self._session.pop_last_presentation(user_id)
+                                if presentation is not None:
+                                    await self.render_presentation(
+                                        str(update.message.chat_id), presentation
+                                    )
+                            except Exception as e:  # pragma: no cover — defensive
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)